vizro-core/tests/unit/vizro/models/_action/test_action.py

```diff
@@ -1,45 +1,99 @@
 """Unit tests for vizro.models.Action."""
 
-import sys
+import textwrap
 
-import pandas as pd
 import pytest
 from asserts import assert_component_equal
 from dash import Output, State, html
 from pydantic import ValidationError
 
 import vizro.models as vm
-import vizro.plotly.express as px
 from vizro import Vizro
-from vizro.actions import export_data
-from vizro.managers import model_manager
+from vizro.actions import filter_interaction
 from vizro.models._action._action import Action
 from vizro.models.types import capture
 
 
 @pytest.fixture
-def custom_action_function_mock_return(request):
+def custom_action_mock_return(request):
+    """Return an action function that returns the value of the request param."""
+
     @capture("action")
-    def _custom_action_function_mock_return():
+    def _custom_action_mock_return():
         return request.param
 
-    return _custom_action_function_mock_return
+    return _custom_action_mock_return
 
 
 @pytest.fixture
-def custom_action_build_expected():
-    return html.Div(id="action_test_action_model_components_div", children=[], hidden=True)
+def custom_action_with_parameters(request):
+    """Return a custom action function with the specified parameters."""
+    # Parameters in format: "arg_1, arg_2, arg_3"
+    parameters = ", ".join(request.param)
+
+    # Create a function with the specified number of parameters that returns their values
+    func_code = textwrap.dedent(f"""
+        from vizro.models.types import capture
+
+        @capture("action")
+        def custom_action({parameters}):
+            pass
+    """)
+
+    local_vars = {}
+    exec(func_code, {}, local_vars)
+
+    return local_vars["custom_action"]
```

Comment:
This is very clever but I think we should avoid doing it if at all possible 😅 Can we instead just do just by defining 3 separate functions for 0, 1 and 2 arguments? Those can still be passed in as `pytest.mark.parametrize`.